inc/tag_replacer.php

```diff
@@ -62,22 +62,97 @@ public function init() {
 	 *
 	 * @return mixed
 	 */
+
+	/**
+	 *  Replace in given content the given image tags with video tags is image is gif
+	 *
+	 * @param string $image_url Image url to process.
+	 * @param string $image_tag The tag to replace.
+	 * @param string $content The content to process.
+	 *
+	 * @return bool
+	 */
+	public function img_to_video( $image_url, $image_tag, &$content ) {
+		if ( $this->settings->get( 'img_to_video' ) === 'disabled' ) {
+			return false;
+		}
+
+		$file = get_headers( $image_url, 1 );
```

Comment:
we should avoid checking the content type in this way, checking the extension only is enough.